src/nomad_simulations/schema_packages/model_method.py

```diff
@@ -1345,6 +1345,144 @@ class BSE(ExcitedStateMethodology):
+    type = Quantity(
+        type=MEnum('linear_response', 'real_time'),
+        default='linear_response',
+        description="""
+        TDDFT flavour (linear-response or real-time propagation).
+          • linear_response — frequency-domain (Casida, Sternheimer, Liouville-Lanczos)
+          • real_time       — explicit time propagation under a perturbation.
+        """,
+    )
+
+    kernel = Quantity(
+        type=MEnum(
+            'ALDA',
+            'AGGA',
+            'hybrid',
+            'range_separated',
+            'frequency_dependent',
+            'bootstrap',
+            'long_range_corrected',
+        ),
+        description="""
+        This field describes the effective XC response behaviour assumed in the TDDFT calculation, 
+        whether implemented explicitly (linear-response) or implicitly via the XC potential (real-time).
+          • ALDA / AGGA     : adiabatic local or semi-local
+          • hybrid          : hybrid adiabatic kernel
+          • range_separated : long/short-range split hybrids
+          • frequency_dependent : non-adiabatic kernel
+          • bootstrap / long_range_corrected : common nano/solid-state kernels
+        """,
+    )
+
+    is_tamm_dancoff = Quantity(
+        type=bool,
+        description='If True, Tamm-Dancoff approximation (TDA) is used (linear-response only).',
+    )
+
+    solver = Quantity(
+        type=MEnum(
+            'Casida', 'Sternheimer', 'Liouville-Lanczos', 'propagation', 'unavailable'
+        ),
+        description="""
+        Numerical driver:
+          • Casida             : eigenvalue problem in particle-hole space
+          • Sternheimer        : frequency-domain Sternheimer
+          • Liouville-Lanczos  : iterative Liouville-Lanczos
+          • propagation        : explicit time evolution (real-time)
+          • unavailable        : code-chosen / unspecified
+        """,
+    )
+
+    xc_kernel_ref = Quantity(
+        type=XCFunctional,
+        description="""
+        Reference to an `XCFunctional` describing the TDDFT kernel.
+        Typically points to the ground-state DFT XC section if the adiabatic
+        kernel matches, or to a separate XCFunctional created for the TD kernel
+        when it differs (e.g., LRC/hybrid kernel on top of semi-local ground state).
+        """,
+    )
+
+    field_ref = Quantity(
+        type=Photon,
+        description='External field / polarization used to drive the response or propagation.',
+    )
+
+    target = Quantity(
+        type=MEnum('absorption', 'emission', 'EELS', 'Raman', 'nonlinear'),
+        description='Intended spectral/response target of the TDDFT input.',
+    )
+
+    time_step = Quantity(
+        type=np.float64,
+        unit='second',
+        description='Time step used for real-time propagation.',
+    )
+
+    n_steps = Quantity(
+        type=np.int32,
+        description='Number of time steps in real-time propagation.',
+    )
+
+    propagator = Quantity(
```

Comment:
same comment as for `time_step`

this is also related with numerical_settings.Mesh (cc @ndaelman-hu)